### ISSUE TITLE
Pass through underlying errors for failing capabilities

### ISSUE THREAD

**Issue by martnst** [ProcedureKit/ProcedureKit#513]
Somehow related to #512 but not about naming. 

Working with the Capabilities and my error handling I notice that in case a `AuthorizedFor` condition fails it returns an `ProcedureKitError.capabilityUnauthorized()` error. 

In the case of requiring access to CloudKit there might occur errors which have no way out though the current interface of `AuthorizationStatusProtocol.meets(requirement:) -> Bool`. Maybe you can improve this here to pass the underlying error though and either return it instead of or nesting it in the `ProcedureKitError.capabilityUnauthorized()`?

**Comment by danthorpe** [comment]
Yep, at the moment, I've not moved all the capabilities across. And for the CloudKit one especially it is a little bit involved. When I work on it, the `func meets(requirement: Requirement) -> Bool` may well become a throwing function.
